server/src/domain/media/media.service.ts

```diff
@@ -59,51 +58,89 @@ export class MediaService {
-        if (!mainVideoStream) {
-          this.logger.error(`Could not extract thumbnail for asset ${asset.id}: no video streams found`);
+        try {
+          await this.generateVideoThumbnail(asset, 'jpeg');
+          this.logger.log(`Successfully generated JPEG video thumbnail ${asset.id}`);
+        } catch (err) {
+          this.logger.error(`Could not generate JPEG video thumbnail for asset ${asset.id}: ${err}`);
           return false;
         }
-        const { ffmpeg } = await this.configCore.getConfig();
-        const config = { ...ffmpeg, targetResolution: thumbnail.jpegSize.toString(), twoPass: false };
-        const options = new ThumbnailConfig(config).getOptions(mainVideoStream);
-        await this.mediaRepository.transcode(asset.originalPath, jpegThumbnailPath, options);
-        this.logger.log(`Successfully generated video thumbnail ${asset.id}`);
         break;
+      default:
+        throw new UnsupportedMediaTypeException(`Unsupported asset type for thumbnail generation: ${asset.type}`);
     }
 
-    await this.assetRepository.save({ id: asset.id, resizePath: jpegThumbnailPath });
-
+    const resizePath = this.getPath(asset, 'jpeg');
+    await this.assetRepository.save({ id: asset.id, resizePath });
     return true;
   }
 
+  async generateImageThumbnail(asset: AssetEntity, format: 'jpeg' | 'webp') {
+    const { thumbnail } = await this.configCore.getConfig();
+    const size = format === 'jpeg' ? thumbnail.jpegSize : thumbnail.webpSize;
+    const thumbnailOptions = { format, size, wideGamut: thumbnail.wideGamut, quality: thumbnail.quality };
+    const thumbnailPath = this.getPath(asset, format);
+    await this.mediaRepository.resize(asset.originalPath, thumbnailPath, thumbnailOptions);
+  }
+
+  async generateVideoThumbnail(asset: AssetEntity, format: 'jpeg' | 'webp') {
+    const { ffmpeg, thumbnail } = await this.configCore.getConfig();
+    const size = format === 'jpeg' ? thumbnail.jpegSize : thumbnail.webpSize;
+    const { videoStreams } = await this.mediaRepository.probe(asset.originalPath);
+    const mainVideoStream = this.getMainVideoStream(videoStreams);
+    if (!mainVideoStream) {
+      throw new UnsupportedMediaTypeException(
+        `Could not extract thumbnail for asset ${asset.id}: no video streams found`,
+      );
+    }
+    const thumbnailPath = this.getPath(asset, format);
+    const config = { ...ffmpeg, targetResolution: size.toString() };
+    const options = new ThumbnailConfig(config).getOptions(mainVideoStream);
+    await this.mediaRepository.transcode(asset.originalPath, thumbnailPath, options);
+  }
+
   async handleGenerateWebpThumbnail({ id }: IEntityJob) {
     const [asset] = await this.assetRepository.getByIds([id]);
     if (!asset || !asset.resizePath) {
       return false;
     }
 
-    const webpPath = asset.resizePath.replace('jpeg', 'webp').replace('jpg', 'webp');
+    switch (asset.type) {
+      case AssetType.IMAGE:
+        try {
```

Comment:
Why duplicate this try? I would wrap the switch in it, or omit it entirely. I think you can await something and use .catch for the log?

Reply:
You can't wrap the whole thing in a try because that would also catch `UnsupportedMediaTypeException`.

Reply:
There just be ways to clean it up though. Even just moving logging elsewhere or merging log lines would help?

So instead of a successful log in two places, it could be moved out of the switch statement entirely and the asset kind formatted into the string. These logs also don't really provide any useful info other than just logging, so maybe the log could be printed downstream?

Reply:
There actually is already a try/catch upstream that runs and logs whenever a job throws an unhandled error. We can probably just remove these.